api/src/test/java/org/openmrs/module/eptsreports/reporting/intergrated/utils/resultsmatching/ResultsMatchingTest.java

```diff
@@ -0,0 +1,221 @@
+                  + INDICATOR_MAPPINGS_FILE_NAME);
+    } else {
+      indicatorMappingFile =
+          new File(getClass().getClassLoader().getResource(INDICATOR_MAPPINGS_FILE_NAME).getPath());
+    }
+    eptsReportsResultsMatchingConfig =
+        (JSONObject) new JSONParser().parse(new FileReader(indicatorMappingFile));
+    Context.authenticate(
+        (String) ((JSONObject) eptsReportsResultsMatchingConfig.get("user")).get("name"),
+        (String) ((JSONObject) eptsReportsResultsMatchingConfig.get("user")).get("pass"));
+    reportDefinitionService = Context.getService(ReportDefinitionService.class);
+  }
+
+  @Test
+  public void performTest() throws Exception {
+    ReportUtil.updateGlobalProperty(
+        ReportingConstants.GLOBAL_PROPERTY_DATA_EVALUATION_BATCH_SIZE, "-1");
+    ReportUtil.updateGlobalProperty(ReportingConstants.DEFAULT_LOCALE_GP_NAME, "en");
+
+    boolean missMatch = false;
+    List<RunResult> eptsReportTestResults = new ArrayList<RunResult>();
+    Iterator mappingConfigIterator =
+        ((JSONArray) eptsReportsResultsMatchingConfig.get("indicatorMappings")).iterator();
+    while (mappingConfigIterator.hasNext()) {
+      JSONObject reportConfig = (JSONObject) mappingConfigIterator.next();
+
+      // the first and only key is uuid
+      String reportUuid = reportConfig.keySet().iterator().next().toString();
+      JSONObject mappingsObject = (JSONObject) reportConfig.get(reportUuid);
+      JSONObject parameterValues = (JSONObject) mappingsObject.get("parameterValues");
+      EvaluationContext context = new EvaluationContext();
+      context.addParameterValue(
+          "startDate", DATE_FORMAT.parse((String) parameterValues.get("startDate")));
+      context.addParameterValue(
+          "endDate", DATE_FORMAT.parse((String) parameterValues.get("endDate")));
+      context.addParameterValue(
+          "location",
+          Context.getLocationService()
+              .getLocation(Integer.parseInt((String) parameterValues.get("locationId"))));
+
+      ReportDefinition currentReportDefinition =
+          reportDefinitionService.getDefinitionByUuid(reportUuid);
+      StopWatch stopWatch1 = new StopWatch();
+      stopWatch1.start();
+      ReportData currentData = reportDefinitionService.evaluate(currentReportDefinition, context);
+      stopWatch1.stop();
+
+      ReportDefinition masterReportDefinition =
+          reportDefinitionService.getDefinitionByUuid((String) mappingsObject.get("masterReport"));
+      StopWatch stopWatch2 = new StopWatch();
+      stopWatch2.start();
+      ReportData masterData = reportDefinitionService.evaluate(masterReportDefinition, context);
+      stopWatch2.stop();
+
+      List<Match> matches = new ArrayList<Match>();
+      Iterator mappingsIterator = ((JSONArray) mappingsObject.get("mappings")).iterator();
+      // set to "" if none
+      String dataSetKeyAppender = (String) mappingsObject.get("dataSetKeyAppender");
+      while (mappingsIterator.hasNext()) {
```

Comment:
@kaweesi oh one more thing, since you have included all the logic in the same method please include comments describing what each section does. For example, ` Read config`, `Run reports`, `Evaluate Results`...etc..something like that to make it a bit more readable.